AppCenterCrashes/AppCenterCrashes/MSCrashes.mm

```diff
@@ -673,6 +676,9 @@ - (void)configureCrashReporterWithUncaughtExceptionHandlerEnabled:(BOOL)enableUn
 
     // Add a handler for C++-Exceptions.
     [MSCrashesUncaughtCXXExceptionHandlerManager addCXXExceptionHandler:uncaught_cxx_exception_handler];
+
+    // Activate category to handle additional crash details.
+    [MSCrashesCategory activateCategory];
```

Comment:
We need to provide a way for customers to deactivate swizzling but yet with the possibility to use this feature at the cost of some more integration for them. It's a concept adopted by the SDK in other places where we do swizzling.

Reply:
After thinking more about this point, I have more doubts about it.
From our docs:

> App Center uses swizzling to automatically forward the app delegate's methods to App Center services to improve SDK integration. Method swizzling is a way to change the implementation of methods at runtime. There is a possibility of conflicts with other third party libraries or the app delegate itself.

It makes sense to make it configurable for **AppDelegate** to avoid possible conflicts, but this is another case:
* It swizzles NOT delegate's method
* There is a requirement for only one crash reporting library in the project already
* We can't reuse the same config flag
* It's already under another flag and NOT enabled by default

Also, to make it configurable we need to provide a way to do it manually without swizzling. I see two options how:
1. Add **second** implementation of these methods wrapped to custom `NSApplication` class and document how to configure it in `.plist`. This way was used by HockeyApp.
2. Add static public method to `MSCrashes` to forward the exception details from these overrides. (I guess this method will confuse users)

All of above requires adding new public API, so let's discuss it first.

Reply:
This is improving SDK behavior, not providing an optional functionality. It seems odd to have a config to get important information about crashes, I don't think this is the one that we provide a way to opt-in/out.

Reply:
I saw customers just deactivating swizzling simply because of an internal policy. So yes we need this.

Reply:
So, what API/way to do it without swizzling looks best for you?

Reply:
OK so what I would do is try to adapt the patterns we used for swizzling in the SDK 1) for coherency and 2) because it has worked great so far. It means exposing a static method in Crashes to process `reportException:` and `sendEvent:` but it's true that it's a bit confusing and generic so I would rename to `applicationReportException:` and `applicationSendEvent:`. The swizzling will just call these methods under the hood so the processing logic is in only one place.

When Swizzling is deactivated it will be almost as documented by HA. I don't think we need to provide an NSApplication subclass, it will just be for the few apps deactivating swizzling. So that would be:
* Swizzling deactivation should follow the same pattern as usual with the plist boolean for deactivation (i.e.: `AppCenterApplicationForwarderEnabled` set to NO). 
* Then customers subclass NSApplication and update the plist accordingly.
* Implement `sendEvent:` and `applicationSendEvent:` to forward to super implementation and the corresponding methods in Crashes.
* Done.

Swizzling activation should be done in `+load` method for additional stability, I am not super worried about `reportException` but `sendEvent` is probably called more often during the app lifecycle.

Reply:
Almost OK with it except exposing API for `sendEvent:` - only `reportException:` is enough

Reply:
OK fair enough if swizzling + system call to `sendEvent` and `reportException` are guaranteed to all run in main thread then there shouldn't be any problem yes.